### PR TITLE
Better class code errors

### PR DESCRIPTION
**Pull request by jaredsilver**
Addresses Peter's comment on issue #1651 to make error messages a little more specific

### CODE REVIEW

File: client/app/bundles/HelloWorld/components/student_profile/students_classrooms/join_class_stage1.jsx [hunk 1]
```diff
@@ -14,14 +14,18 @@ export default React.createClass({
       .done(function(){
         that.props.advanceStage();
       })
-      .fail(function() {
-        that.setState({error: 'Invalid Classcode'});
+      .fail(function(jqXHR, textStatus, errorThrown) {
```
Comment (RyanNovas) [inline client/app/bundles/HelloWorld/components/student_profile/students_classrooms/join_class_stage1.jsx]: No need to pass `textStatus` or `errorThrown` as we are only relying on `jqXHR`

File: spec/features/students/add_class_spec.rb [hunk 2]
```diff
@@ -47,12 +47,23 @@
...
 
-      it 'displays an error message' do
+      it 'displays the standard error message' do
         page.find(".class-input").set('not-a-class')
         click_button('Join Your Class')
-        eventually {expect(page).to have_content("Invalid Classcode")}
+        eventually {expect(page).to have_content("Oops! Looks like that isn't a valid class code. Please try again.")}
+      end
+
+      it 'displays the archived class error message' do
```
Comment (RyanNovas) [inline spec/features/students/add_class_spec.rb]: Lines 59 to 63 can be refactored into one line with `Classroom.create(name: 'Archived Class, visible: true....`

**Comment by jaredsilver** [comment]
Thanks, @RyanNovas! Just made those adjustments 😃 

I'm super into edtech, and I came across Quill on LearnLaunch's website looking for some ways to get more involved in the Boston edtech scene while studying entrepreneurship at Babson College. I'm a self-taught developer with some side projects in edtech, and I recently interned doing Rails work for an awesome company called [Higher Learning Technologies](http://hltcorp.com).

I hope to make some contributions here (even if they're fairly insignificant things like better error messages) because I find nothing more fulfilling than knowing students are better off thanks to something I contributed to in some way. Also, with that being said, please feel free to let me know at any point if I'm taking up too much of your time with subpar PRs you have to review or anything like that 😃 

Thanks for all the awesome work you're doing!

- Jared

**Comment by RyanNovas** [comment]
@jaredsilver great, thanks! I'm glad that you are helping us. With 10,000 students a day and two engineers we can definitely use the extra dev firepower.

If you wanna keep hacking on Quill (which would be awesome), I'd suggest you join our [Slack Channel](https://quill-slackin.herokuapp.com/) where you could any code questions in realtime.

We also have a list of projects that may be relatively straight forward for those just dipping their toes in the code base here: https://github.com/empirical-org/Empirical-Core/issues?q=is%3Aopen+is%3Aissue+milestone%3A%22Open+Source+Projects%22 

However, feel free to go after whatever seems most appealing to you.

**Comment by ddmck** [comment]
Merging! Thanks @jaredsilver

**Pull request merged**